docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+  by arguments which end with ``_fn`` (short for filename) which refer to a source in the data catalog
+  based on the source name or a file based on the (relative) path to the file. Within a model method the data is read
+  by calling any ``DataCatalog.get_<data_type>`` method which work for both source and file names.
+- The region and res (resolution) arguments used in the command line build and clip methods are passed to the model method referred
+  in the internal *_CLI_ARGS* model constant, which by default in the Model class, is the setup_basemaps method for both arguments. This is
+  typically the first model method which should be called when building a model or your base / region setup method.
+- Workflows define (partial) transformations of data from input data to model data. And should, if possible, be kept generic to be
+  shared between model plugins. The input data is passed to the workflow by python data objects consistent with its associated data
+  types (e.g. xarray.Dataset for regular rasters) and not read by the workflow itself.
+
+.. figure:: ../_static/model_building_process.png
+
+  Schematic of model building process with HydroMT
+
+.. _plugin_setup:
+
+Setup methods
+^^^^^^^^^^^^^
+
+In general, a HydroMT ``setup_<>`` method does 4 things:
+
+  1. read or otherwise fetch data using the ``DataCatalog`` and ``DataAdapter`` and the corresponding ``get_data`` method (
+  ``get_rasterdataset`` for RasterDataset, ``get_GeoDataset`` for GeoDataset, ``get_geodataframe`` for GeoDataFrame and
+  ``get_dataframe`` for DataFrame).
+
+  2. process that data in some way, optionally by calling an external workflow function.
+  3. rename or update attributes from HydroMT variable conventions (name, unit) to the specific model conventions.
+  4. adds the data to the corresponding HydroMT model components.
+
+A simplified example of what a setup function would look like for example to add a landuse grid from a raster input
+data using an external workflow from hydromt core is:
+
+.. code-block:: python
+
+  def setup_landuse(
+        self,
+        landuse_fn: Union[str, Path, xr.DataArray],
+    ):
+        """Add landuse data variable to grid.
+
+        Adds model layers:
+
+        * **landuse_class** grid: data from landuse_fn
+
+        Parameters
+        ----------
+        landuse_fn: str, Path, xr.DataArray
+            Data catalog key, path to raster file or raster xarray data object.
+            If a path to a raster file is provided it will be added
+            to the data_catalog with its name based on the file basename without
+            extension.
+        """
+        self.logger.info(f"Preparing landuse data from raster source {landuse_fn}")
+        # 1. Read landuse raster data
+        da_landuse = self.data_catalog.get_rasterdataset(
+            landuse_fn,
+            geom=self.region,
+            buffer=2,
+            variables=["landuse"],
```

Comment:
check naming convention. I think we often use `lulc`

Reply:
landuse is commonly used in the datacatalog, the only reference to `lulc` I could find was in hydromt/cli/main.py:
https://github.com/Deltares/hydromt/blob/effbf2f15e306e65e397969426d09c8d93912a8e/hydromt/cli/main.py#L273
Not sure which one you'd want